File: docs-old-1103-2/data-tools/codesnippet/VisualBasic/handle-a-concurrency-exception_1.vb
    Private Sub UpdateDatabase()

        Try
            Me.CustomersTableAdapter.Update(Me.NorthwindDataSet.Customers)
            MsgBox("Update successful")

        Catch dbcx As Data.DBConcurrencyException
            Dim response As Windows.Forms.DialogResult

            response = MessageBox.Show(CreateMessage(CType(dbcx.Row, NorthwindDataSet.CustomersRow)),
                "Concurrency Exception", MessageBoxButtons.YesNo)

            ProcessDialogResult(response)

        Catch ex As Exception
            MsgBox("An error was thrown while attempting to update the database.")
        End Try
    End Sub

File: docs/data-tools/codesnippet/CSharp/handle-a-concurrency-exception_1.cs
        private void UpdateDatabase()
        {
            try
            {
                this.customersTableAdapter.Update(this.northwindDataSet.Customers);
                MessageBox.Show("Update successful");
            }
            catch (DBConcurrencyException dbcx)
            {
                DialogResult response = MessageBox.Show(CreateMessage((NorthwindDataSet.CustomersRow)
                    (dbcx.Row)), "Concurrency Exception", MessageBoxButtons.YesNo);

                ProcessDialogResult(response);
            }
            catch (Exception ex)
            {
                MessageBox.Show("An error was thrown while attempting to update the database.");
            }
        }